commit 115f7170d74598b1b96921000af4eef5ed330ea5
Author: kikadf <kikadf.01@gmail.com>
Date:   Fri Apr 10 13:36:34 2026 +0200

    Remove llvm-19 fixes

diff --git a/chrome/browser/net/profile_network_context_service.cc b/chrome/browser/net/profile_network_context_service.cc
index 78b293e430..1fa5f3d35a 100644
--- a/chrome/browser/net/profile_network_context_service.cc
+++ b/chrome/browser/net/profile_network_context_service.cc
@@ -1306,7 +1306,7 @@ std::vector<uint8_t>
 ProfileNetworkContextService::GetEncryptedCachePrimaryKey() {
   std::string encoded_encrypted_primary_key = profile_->GetPrefs()->GetString(
       enterprise_connectors::kEncryptedCachePrimaryKey);
-  return base::Base64Decode(encoded_encrypted_primary_key).value_or(std::vector<uint8_t>{});
+  return base::Base64Decode(encoded_encrypted_primary_key).value_or({});
 }
 
 #endif  // BUILDFLAG(ENTERPRISE_CACHE_ENCRYPTION)
diff --git a/components/autofill/core/browser/payments/full_card_request.cc b/components/autofill/core/browser/payments/full_card_request.cc
index 3322486613..149c2aa241 100644
--- a/components/autofill/core/browser/payments/full_card_request.cc
+++ b/components/autofill/core/browser/payments/full_card_request.cc
@@ -124,7 +124,7 @@ void FullCardRequest::GetFullCardImpl(
   request_->card = std::move(card);
   request_->last_committed_primary_main_frame_origin =
       last_committed_primary_main_frame_origin;
-  request_->context_token = std::move(context_token).value_or(std::string{});
+  request_->context_token = std::move(context_token).value_or({});
   request_->selected_challenge_option = std::move(selected_challenge_option);
 
   should_unmask_card_ = request_->card.masked() ||
diff --git a/components/enterprise/client_certificates/core/private_key_factory.cc b/components/enterprise/client_certificates/core/private_key_factory.cc
index 87985bd1f9..7f94445006 100644
--- a/components/enterprise/client_certificates/core/private_key_factory.cc
+++ b/components/enterprise/client_certificates/core/private_key_factory.cc
@@ -123,8 +123,8 @@ void PrivateKeyFactoryImpl::OnPrivateKeyCreated(
     scoped_refptr<PrivateKey> private_key) {
   if (!private_key && source != PrivateKeySource::kSoftwareKey) {
     for (auto fallback_source =
-             (std::find(std::begin(kKeySourcesOrderedBySecurity),
-                         std::end(kKeySourcesOrderedBySecurity), source)+1);
+             ++std::find(std::begin(kKeySourcesOrderedBySecurity),
+                         std::end(kKeySourcesOrderedBySecurity), source);
          fallback_source != std::end(kKeySourcesOrderedBySecurity);
          fallback_source++) {
       auto it = sub_factories_.find(*fallback_source);
diff --git a/components/named_mojo_ipc_server/named_mojo_server_endpoint_connector_linux.cc b/components/named_mojo_ipc_server/named_mojo_server_endpoint_connector_linux.cc
index 20c03cae12..33a7b409d9 100644
--- a/components/named_mojo_ipc_server/named_mojo_server_endpoint_connector_linux.cc
+++ b/components/named_mojo_ipc_server/named_mojo_server_endpoint_connector_linux.cc
@@ -132,7 +132,7 @@ bool NamedMojoServerEndpointConnectorLinux::TryStart() {
   if (!options_.require_same_peer_user) {
     // Allow any user to write to the UDS. fchmod doesn't work after bind(), so
     // we need to call chmod on the socket filename, which is the server name.
-    if (chmod(options_.server_name.c_str(), 0666) != 0) {
+    if (chmod(options_.server_name.c_str(), 0o666) != 0) {
       PLOG(ERROR) << "chmod failed";
       return false;
     }
diff --git a/media/gpu/vaapi/vaapi_jpeg_encoder.cc b/media/gpu/vaapi/vaapi_jpeg_encoder.cc
index ac3c49e6b0..e87494c736 100644
--- a/media/gpu/vaapi/vaapi_jpeg_encoder.cc
+++ b/media/gpu/vaapi/vaapi_jpeg_encoder.cc
@@ -98,24 +98,24 @@ void FillHuffmanTableParameters(
 
     // Load DC Table.
     auto num_dc_codes = base::span(huffman_tables[i].num_dc_codes);
-//    static_assert(num_dc_codes.size() == dcTable.code_length.size());
+    static_assert(num_dc_codes.size() == dcTable.code_length.size());
     num_dc_codes.copy_from_nonoverlapping(dcTable.code_length);
     // |code_values| of JpegHuffmanTable needs to hold DC and AC code values
     // so it has different size than
     // |huff_table_param->huffman_table[i].dc_values|. Therefore we can't use
     // SafeArrayMemcpy() here.
     auto dc_values = base::span(huffman_tables[i].dc_values);
-//    static_assert(dc_values.size() <= dcTable.code_value.size());
+    static_assert(dc_values.size() <= dcTable.code_value.size());
     dc_values.copy_from_nonoverlapping(
         base::span(dcTable.code_value).first(dc_values.size()));
 
     // Load AC Table.
     auto num_ac_codes = base::span(huffman_tables[i].num_ac_codes);
-//    static_assert(num_ac_codes.size() == acTable.code_length.size());
+    static_assert(num_ac_codes.size() == acTable.code_length.size());
     num_ac_codes.copy_from_nonoverlapping(acTable.code_length);
 
     auto ac_values = base::span(huffman_tables[i].ac_values);
-//    static_assert(ac_values.size() == acTable.code_value.size());
+    static_assert(ac_values.size() == acTable.code_value.size());
     ac_values.copy_from_nonoverlapping(acTable.code_value);
 
     std::ranges::fill(huffman_tables[i].pad, 0);
diff --git a/mojo/public/cpp/bindings/clone_traits.h b/mojo/public/cpp/bindings/clone_traits.h
index 74cd0996dc..1cb9629761 100644
--- a/mojo/public/cpp/bindings/clone_traits.h
+++ b/mojo/public/cpp/bindings/clone_traits.h
@@ -30,12 +30,12 @@ struct CloneTraits {
     } else if constexpr (std::copyable<T>) {
       return input;
     } else {
-/*      static_assert(
+      static_assert(
           false,
           "T is not copyable and has no Clone() method, so the default "
           "mojo::CloneTraits cannot be used; please make sure to include the "
           "header that defines the mojo::CloneTraits<T> specialization");
-*/    }
+    }
   }
 };
 
diff --git a/services/network/public/cpp/permissions_policy/origin_with_possible_wildcards.cc b/services/network/public/cpp/permissions_policy/origin_with_possible_wildcards.cc
index d1b177ebca..e14d79aa57 100644
--- a/services/network/public/cpp/permissions_policy/origin_with_possible_wildcards.cc
+++ b/services/network/public/cpp/permissions_policy/origin_with_possible_wildcards.cc
@@ -94,7 +94,7 @@ bool OriginWithPossibleWildcards::DoesMatchOrigin(
 }
 
 bool operator==(const OriginWithPossibleWildcards& lhs,
-                const OriginWithPossibleWildcards& rhs) { return lhs.csp_source == rhs.csp_source; }
+                const OriginWithPossibleWildcards& rhs) = default;
 
 std::strong_ordering operator<=>(const OriginWithPossibleWildcards& lhs,
                                  const OriginWithPossibleWildcards& rhs) {
diff --git a/third_party/blink/renderer/modules/xr/xr_rigid_transform.h b/third_party/blink/renderer/modules/xr/xr_rigid_transform.h
index a3752ac89e..0049b98bc7 100644
--- a/third_party/blink/renderer/modules/xr/xr_rigid_transform.h
+++ b/third_party/blink/renderer/modules/xr/xr_rigid_transform.h
@@ -12,7 +12,9 @@
 #include "third_party/blink/renderer/modules/modules_export.h"
 #include "third_party/blink/renderer/platform/bindings/script_wrappable.h"
 
-#include "ui/gfx/geometry/transform.h"
+namespace gfx {
+class Transform;
+}
 
 namespace blink {
 
diff --git a/ui/gfx/paint_vector_icon.cc b/ui/gfx/paint_vector_icon.cc
index 710f070b01..eef47b4c9b 100644
--- a/ui/gfx/paint_vector_icon.cc
+++ b/ui/gfx/paint_vector_icon.cc
@@ -235,7 +235,7 @@ void PaintPath(Canvas* canvas,
         break;
 
       case H_LINE_TO: {
-        const SkPoint last_point = path.getLastPt().value_or(SkPoint{0, 0});
+        const SkPoint last_point = path.getLastPt().value_or({0, 0});
         path.lineTo(arg(0), last_point.fY);
         break;
       }
@@ -245,7 +245,7 @@ void PaintPath(Canvas* canvas,
         break;
 
       case V_LINE_TO: {
-        const SkPoint last_point = path.getLastPt().value_or(SkPoint{0, 0});
+        const SkPoint last_point = path.getLastPt().value_or({0, 0});
         path.lineTo(last_point.fX, arg(0));
         break;
       }
@@ -273,7 +273,7 @@ void PaintPath(Canvas* canvas,
         // details.
         // Note that |x1| and |y1| will correspond to the sole control point if
         // calculating a quadratic curve.
-        const SkPoint last_point = path.getLastPt().value_or(SkPoint{0, 0});
+        const SkPoint last_point = path.getLastPt().value_or({0, 0});
         SkScalar delta_x = 0;
         SkScalar delta_y = 0;
         if (IsCommandTypeCurve(previous_command_type)) {
